pkgs/by-name/ob/obj2voxel/package.nix

```diff
@@ -0,0 +1,75 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, fetchpatch
+, static ? stdenv.hostPlatform.isStatic
+, config
+, cmake
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "obj2voxel";
+  version = "1.3.4";
+
+  src = (
+    fetchFromGitHub {
+      owner = "Eisenwave";
+      repo = finalAttrs.pname;
+      rev = "refs/tags/v${finalAttrs.version}";
+      fetchSubmodules = true;
+      hash = "sha256-FkGQjYm15le1Na+2U1TgK4nNOZNs5bME5J87pjVFpDs=";
+    }
+  ).overrideAttrs (
+    _: {
+      GIT_CONFIG_COUNT = 1;
+      GIT_CONFIG_KEY_0 = "url.https://github.com/.insteadOf";
+      GIT_CONFIG_VALUE_0 = "git@github.com:";
+    }
+  );
+
+  patches = [
+    # pull missing definitions
+    (
+      fetchpatch {
+        name = "add-std-includes.patch";
+        url = "https://patch-diff.githubusercontent.com/raw/Eisenwave/voxel-io/pull/3.diff";
+        hash = "sha256-2JeFBx0Ds++BplWXuxOo4KOAW6gXCT/kFI74i0M0ROM=";
+        stripLen = 1;
+        extraPrefix = "voxelio/";
+      }
+    )
+    # fix compile error on aarch64-darwin
+    (
+      fetchpatch {
+        name = "fix-aarch64-darwin-type-mismatch.patch";
+        url = "https://patch-diff.githubusercontent.com/raw/Eisenwave/voxel-io/pull/4.diff";
+        hash = "sha256-AuvjpTGBLKN0f47IoxIDkV32g9A/4RbvGBo3ToGiH/w=";
+        stripLen = 1;
+        extraPrefix = "voxelio/";
+      }
+    )
+    # already fixed in upstream git head
+    ./1.3.4-fix-type-mismatch.patch
```

Comment:
We can't use fetchpatch?

Reply:
IMHO, not easily. By now, upstream master avoids the issue because all these `size_t` typed variables have been changed to `usize`. But this is a rather global change not easily applied, so I decided towards a clear and small patch that just solves the issue remaining after applying the fixes that made it into pull requests.

IMHO, the next best solution would actually be to just build the upstream master. But for now, I decided to keep with the usual policy of building something that has a version number.